gensim/models/word2vec.py

```diff
@@ -474,22 +503,27 @@ def create_binary_tree(self):
 
             logger.info("built huffman tree with maximum node depth %i", max_depth)
 
-    def build_vocab(self, sentences, keep_raw_vocab=False):
+    def build_vocab(self, sentences, keep_raw_vocab=False, trim_rule=None):
         """
         Build vocabulary from a sequence of sentences (can be a once-only generator stream).
         Each sentence must be a list of unicode strings.
 
         """
-        self.scan_vocab(sentences)  # initial survey
-        self.scale_vocab(keep_raw_vocab)  # trim by min_count & precalculate downsampling
+        if trim_rule is not None:
```

Comment:
A doc2vec test fails since doc2vec uses this method. The alternative is to add the rule parameter to doc2vec as well, but I thought that would be best left to another PR.